src/translators/sql_translator.py

```diff
@@ -455,8 +455,175 @@ def is_text(attr_type):
+                    # so use None which will be inserted as NULL to the db.
+                    values.append(None)
+                except ValueError:
+                    # this value cannot be cast to column type
+                    # so use None which will be inserted as NULL to the db.
+                    values.append(None)
+        return values
+
+    @staticmethod
+    def _is_ngsi_array(attr, attr_t):
+        return (attr_t == NGSI_STRUCTURED_VALUE
+                and 'value' in attr and isinstance(attr['value'], list)) or attr_t == "Array"
+
+    @staticmethod
+    def _is_ngsi_object(attr, attr_t):
+        return attr_t == NGSI_STRUCTURED_VALUE or ('value' in attr and isinstance(attr['value'], dict))
+
+    @staticmethod
+    def _is_ngsi_ld_datetime_property(attr):
+        if 'type' in attr and attr[
+            'type'] == 'Property' and 'value' in attr and isinstance(
+                attr['value'], dict) \
+                and '@type' in attr['value'] and attr['value'][
+            '@type'] == 'DateTime':
+            return True
+        return False
+
+    @staticmethod
+    def _ngsi_geojson_to_db(attr):
         raise NotImplementedError
 
+    @staticmethod
+    def _ngsi_number_to_db(attr):
+        try:
+            if isinstance(attr['value'], bool):
+                return None
+            elif isinstance(attr['value'], float):
+                return attr['value']
+            elif attr['value'] is not None:
+                return float(attr['value'])
+        except ValueError:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+
+
+    @staticmethod
+    def _ngsi_datetime_to_db(attr):
+        if SQLTranslator._is_iso_date(attr['value']):
+            return attr['value']
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+
+    @staticmethod
+    def _ngsi_integer_to_db(attr):
+        try:
+            if isinstance(attr['value'], bool):
+                return None
+            elif isinstance(attr['value'], int):
+                return attr['value']
+            elif attr['value'] is not None:
+                return int(float(attr['value']))
+        except ValueError:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+
+    @staticmethod
+    def _ngsi_boolean_to_db(attr):
+        if isinstance(attr['value'], str) and attr['value'].lower() == 'true':
+            return True
+        elif isinstance(attr['value'], str) \
```

Comment:
should we strip whitespace? so e.g. `' false '`, `'false '`, etc. get converted to `False`? If so then we could just add a trailing call to `strip` as in `attr['value'].lower().strip()`.